basic_pitch/inference.py

```diff
@@ -90,11 +230,15 @@ def get_audio_input(
 
     original_length = audio_original.shape[0]
     audio_original = np.concatenate([np.zeros((int(overlap_len / 2),), dtype=np.float32), audio_original])
-    audio_windowed, window_times = window_audio_file(audio_original, hop_size)
-    return audio_windowed, window_times, original_length
+    for window, window_time in window_audio_file(audio_original, hop_size):
+        yield np.expand_dims(window, axis=0), window_time, original_length
```

Comment:
what is the rationale behind the switch to yield over return here?

Reply:
Instead of returning a list of items and allocating all the memory for that, yield lets us return just whats needed when its needed.